src/lib/crypto/kimchi_bindings/stubs/src/pasta_fp_plonk_proof.rs

```diff
@@ -295,13 +294,17 @@ pub fn caml_pasta_fp_plonk_proof_example_with_foreign_field_mul(
     let foreign_field_modulus = Fq::modulus_biguint();
 
     // Layout
-    //      0    ForeignFieldMul   (foreign field multiplication gadget)
-    //      1    Zero              (foreign field multiplication gadget)
-    //      4-7  multi-range-check (left multiplicand)
-    //      8-11 multi-range-check (right multiplicand)
-    //     12-15 multi-range-check (product1_lo, product1_hi_0, carry1_lo)
-    //     16-19 multi-range-check (result range check)
-    //     20-23 multi-range-check (quotient range check)
+    //      0-1  ForeignFieldMul | Zero
```

Comment:
I find it difficult to line up the code below with this witness structure.  It's a complicated one.  Perhaps some kind of key noted in the comments could help.  For example...

```
//    L1      2-5  compact-multi-range-check (result range check)
//    L2     6  "single" Generic (result bound)
//    L2    7-10 multi-range-check (quotient range check)
```

and then below in the code write comments like

// L1: blah blah...